server/workflow/impl/Packrat/WorkflowUpload.ts

```diff
@@ -103,12 +105,18 @@ export class WorkflowUpload implements WF.IWorkflow {
             this.appendToWFReport(`Upload validation of ${RSR.fileName}`);
 
             let fileRes: H.IOResults = { success: true };
-            if (isModel) // if we're a model, zipped or not, validate the entire file/collection as is:
+            if (isModel) {
+                // if we're a model, zipped or not, validate the entire file/collection as is:
                 fileRes = await this.validateFileModel(RSR.fileName, RSR.readStream, false, idSystemObject);
-            else if (path.extname(RSR.fileName).toLowerCase() !== '.zip') // not a zip
+            } else if (path.extname(RSR.fileName).toLowerCase() !== '.zip') { // not a zip
+                // we are not a zip
                 fileRes = await this.validateFile(RSR.fileName, RSR.readStream, false, idSystemObject, asset);
-            else {
-                const ZS: ZipStream = new ZipStream(RSR.readStream);
+            } else {
+
+                // it's not a model (e.g. Capture Data)
+                // use ZipFile so we don't need to load it all into memory
+                const filePath: string = Config.storage.rootStaging+'/'+assetVersion.StorageKeyStaging;

```

Comment:
(Same feedback as for PR 543: https://github.com/Smithsonian/dpo-packrat/pull/543#pullrequestreview-1437262814)

Eric, this line of code introduces a hard linkage between the guts of the storage system (i.e. where staged files are stored) and other parts of Packrat. My vision for Packrat was to keep access to storage clearly separated across interface boundaries, which in turn would allow for a future where bits are stored in the cloud (or wherever the storage implementation wants them).

Instead of the approach you've taken here, I recommend using the storage interface to gain access to the staged file. It's been a minute since I looked at this code, but it's definitely possible to do this -- and doing so removes the hard-coded linkage that has been introduced here. I strongly encourage you to dig more deeply here and make use of the storage interface as it was intended to be used.